src/WebApiContrib.Core.Formatter.Csv/CsvFormatterOptions.cs

```diff
@@ -7,5 +7,7 @@ public class CsvFormatterOptions
         public string CsvDelimiter { get; set; } = ";";
 
         public string Encoding { get; set; } = "ISO-8859-1";
+        
+        public bool UseNewtonsoftJsonDataAnnotations { get; set; } = false;

```

Comment:
I'm not convinced this should be an option. if you use JSON.NET, and you have the attributes on your model, you surely expect them to be respected, no?
I think we could always just use them without this opt in need

Reply:
Why not then simply set the default to true? IMO it's always better to let the user decide what he really needs in the end no? 

I understand it adds a little overhead when checking for display name to determine if it should use the JSON.Net attributes, but it's worth it IMO. 

Also, it could be useful to let the end user specify if he wants the display name at all or if he simply wants the property name. This is plausible in an environment where the same class is used both for display purposes in client side app and for data export. There are scenarios where the exported data should not contain display names containing special characters, spaces etc... which make the parsing on the other side a little more complex.

Reply:
yes good point. We only want to use the json ignore and json property, as this is always available. I'm for removing this. @os1r1s110 what's your opinion?

Reply:
See above, we posted simultaneously hehe :) I don't really have a strong opinion on this though, it's just my point of view that the end user should always have as much options as he can (while having sensible defaults previously set to make it less verbose when setting up)

Reply:
if we decide to keep it, we could just rename it to UseJsonPropertyJsonIgnoreAttributes
@filipw @os1r1s110

Reply:
And set the default value to true

Reply:
Seems legit to me. I'd still interested in knowing the reason why @filipw thinks it should not be included, is it because of added overhead when performing the export or something else?

He might have a valid reason for pointing that out.

Reply:
sorry if I wasn't clear. I just meant that we should definitely respect JSON.NET attributes, and that we should always check for them. I don't think there should be a setting to opt-in or opt-out from that behavior.

This is consistent with the MVC behavior in general. You could for example define an API model like this:

```
    public class Person
    {
        [JsonProperty(Required = Required.Always)]
        public string FirstName { get; set; }

        [JsonIgnore]
        public string LastName { get; set; }
    }
```

And rely on the fact that `ModelState` will be invalid if the caller doesn't send the first name. There is no dedicated opt in or opt out from that behavior, as it's implicit on the formatter level.

So for the sake of consistency we should behave the same way.

Reply:
I think you were clear. My main question was more to understand why you preferred removing the option completely. Now I do understand that it is to align with MVC behavior. 

I still personally would prefer leaving the option there and simply set it to true by default so that if the user has efffectively set JsonProperty/JsonIgnore on the model, they will be respected. But for some other cases as the one I mentioned above, it could be nice to leave the option there so that the export can ignore theses properties alltogether and use the basic property name instead for example.

As I also said earlier, that's only my opinion and I don't mind if you choose to remove the option and use the Json Net attributes directly. If I need more options, I can always just copy the code and add it back in my "own" version :)